sql/updates/0.0.9--0.0.10.sql

```diff
@@ -1,5 +1,177 @@
--- these go for good.
+--
+DO $BODY$
+DECLARE
+	pgvector_exists boolean;
+	am_name TEXT;
+  r pg_indexes%ROWTYPE;
+  indexes_cursor REFCURSOR;
+  index_names TEXT[] := '{}';
+  index_definitions TEXT[] := '{}';
+BEGIN
+	-- Function to recreate operator classes for specified access method
+	CREATE OR REPLACE FUNCTION _lantern_internal._recreate_ldb_operator_classes(access_method_name TEXT) RETURNS BOOLEAN AS $$
+	DECLARE
+    	dist_l2sq_ops TEXT;
+    	dist_l2sq_ops_drop TEXT;
+    	dist_cos_ops TEXT;
+    	dist_cos_ops_drop TEXT;
+    	dist_hamming_ops TEXT;
+    	dist_hamming_ops_drop TEXT;
+	BEGIN
+	  	
+    	-- Construct the SQL statement to create the operator classes dynamically.
+    	dist_l2sq_ops_drop := 'DROP OPERATOR CLASS IF EXISTS dist_l2sq_ops USING ' || access_method_name || ' CASCADE;'; 
+    	dist_l2sq_ops := '
+        CREATE OPERATOR CLASS dist_l2sq_ops
+        DEFAULT FOR TYPE real[] USING ' || access_method_name || ' AS
+        OPERATOR 1 <?> (real[], real[]) FOR ORDER BY float_ops,
+        FUNCTION 1 l2sq_dist(real[], real[]),
+        OPERATOR 2 <-> (real[], real[]) FOR ORDER BY float_ops,
+        FUNCTION 2 l2sq_dist(real[], real[]);
+    	';
+    	
+    	dist_cos_ops_drop := 'DROP OPERATOR CLASS IF EXISTS dist_cos_ops USING ' || access_method_name || ' CASCADE;';
+    dist_cos_ops := '
+        CREATE OPERATOR CLASS dist_cos_ops
+        FOR TYPE real[] USING ' || access_method_name || ' AS
+        OPERATOR 1 <?> (real[], real[]) FOR ORDER BY float_ops,
+        FUNCTION 1 cos_dist(real[], real[]),
+        OPERATOR 2 <=> (real[], real[]) FOR ORDER BY float_ops,
+        FUNCTION 2 cos_dist(real[], real[]);
+    ';
+    
+    	
+    	dist_hamming_ops_drop := 'DROP OPERATOR CLASS IF EXISTS dist_hamming_ops USING ' || access_method_name || ' CASCADE;';
+    dist_hamming_ops := '
+        CREATE OPERATOR CLASS dist_hamming_ops
+        FOR TYPE integer[] USING ' || access_method_name || ' AS
+        OPERATOR 1 <?> (integer[], integer[]) FOR ORDER BY float_ops,
+        FUNCTION 1 hamming_dist(integer[], integer[]),
+        OPERATOR 2 <+> (integer[], integer[]) FOR ORDER BY integer_ops,
+        FUNCTION 2 hamming_dist(integer[], integer[]);
+    ';
+
+
+    	-- Execute the dynamic SQL statement.
+    	EXECUTE dist_l2sq_ops_drop;
+    	EXECUTE dist_l2sq_ops;
+    	EXECUTE dist_cos_ops_drop;
+    	EXECUTE dist_cos_ops;
+    	EXECUTE dist_hamming_ops_drop;
+    	EXECUTE dist_hamming_ops;
+
+    	RETURN TRUE;
+	END;
+	$$ LANGUAGE plpgsql VOLATILE;
+
+	-- Check if the vector type from pgvector exists
+	SELECT EXISTS (
+		SELECT 1
+		FROM pg_type
+		WHERE typname = 'vector'
+	) INTO pgvector_exists;
+	
+	am_name := 'hnsw';
+		
+
+	IF pgvector_exists THEN
+		am_name := 'lantern_hnsw';
+    -- these go for good.
+    DROP OPERATOR CLASS IF EXISTS dist_vec_hamming_ops USING hnsw CASCADE;
```

Comment:
should this be `USING lantern_hnsw`?

Reply:
or use `am_name`? It is the code I had written in the last PR and I think I forgot to consider the changed naming of our AM when pgvector is present

Reply:
True, that should be lantern_hnsw;